lib/trie/trie.go

```diff
@@ -303,7 +302,7 @@ func (t *Trie) insert(parent Node, key []byte, value Node) Node {
 		p := newParent.(*node.Leaf)
 		// if a value already exists in the trie at this key, overwrite it with the new value
 		// if the values are the same, don't mark node dirty
-		if p.Value != nil && bytes.Equal(p.Key, key) {
+		if bytes.Equal(p.Key, key) {
```

Comment:
@noot What do you think about this change? Otherwise it would produce a branch from a leaf with nil value. If leaves with nil value are not meant to exist, then we can also remove that condition.

Reply:
a leaf should never have nil value, so this seems fine